packages/trees/backend/sirius-components-collaborative-trees/src/main/java/org/eclipse/sirius/components/collaborative/trees/handlers/CreateTreeEventHandler.java

```diff
@@ -0,0 +1,117 @@
+import reactor.core.publisher.Sinks.Many;
+import reactor.core.publisher.Sinks.One;
+
+/**
+ * Handler used to create a new Tree representation.
+ *
+ * @author Jerome Gout
+ */
+@Service
+public class CreateTreeEventHandler implements IEditingContextEventHandler {
+
+    private final IRepresentationDescriptionSearchService representationDescriptionSearchService;
+
+    private final IRepresentationPersistenceService representationPersistenceService;
+
+    private final TreeCreationService treeCreationService;
+
+    private final IObjectService objectService;
+
+    private final ICollaborativeMessageService messageService;
+
+    private final Counter counter;
+
+    public CreateTreeEventHandler(IRepresentationDescriptionSearchService representationDescriptionSearchService, IRepresentationPersistenceService representationPersistenceService,
+            TreeCreationService treeCreationService, IObjectService objectService, ICollaborativeMessageService messageService, MeterRegistry meterRegistry) {
+        this.representationDescriptionSearchService = Objects.requireNonNull(representationDescriptionSearchService);
+        this.representationPersistenceService = Objects.requireNonNull(representationPersistenceService);
+        this.treeCreationService = Objects.requireNonNull(treeCreationService);
+        this.objectService = Objects.requireNonNull(objectService);
+        this.messageService = Objects.requireNonNull(messageService);
+
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+    }
+
+    @Override
+    public boolean canHandle(IEditingContext editingContext, IInput input) {
+        if (input instanceof CreateRepresentationInput createRepresentationInput) {
+            return this.representationDescriptionSearchService.findById(editingContext, createRepresentationInput.representationDescriptionId())
+                    .filter(TreeDescription.class::isInstance)
+                    .isPresent();
+        }
+        return false;
+    }
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, IInput input) {
+        this.counter.increment();
+
+        String message = this.messageService.invalidInput(input.getClass().getSimpleName(), CreateRepresentationInput.class.getSimpleName());
+        IPayload payload = new ErrorPayload(input.id(), message);
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, editingContext.getId(), input);
+
+        if (input instanceof CreateRepresentationInput createRepresentationInput) {
+            Optional<TreeDescription> optionalDeckDescription = this.representationDescriptionSearchService.findById(editingContext, createRepresentationInput.representationDescriptionId())
+                    .filter(TreeDescription.class::isInstance)
```

Comment:
should be renamed to optionalTreeDescription

Reply:
Done